jsk_arc2017_baxter/euslisp/lib/arc-interface.l

```diff
@@ -586,26 +585,21 @@
           (unless av
             (ros::ros-error "[:pick-object-with-movable-region] arm:~a IK before trying grasp fails. Abort picking" arm)
             (return-from :pick-object-with-movable-region :ik-failed)))
-        ;; Move whole arm to target pose
+        ;; Move whole arm to location above of the target pose
         (send *ri* :angle-vector-sequence-raw next-avs
               :fast (send *ri* :get-arm-controller arm :head t) 0 :scale 5.0)
         (send *ri* :wait-interpolation))
+      ;; Now hand is inside of the bin, so we can use cylindrical
+      (send *ri* :move-hand arm (send *baxter* :hand-grasp-pre-pose arm :cylindrical) 1000)
+      ;; Picking trials in the bin
       (dotimes (i n-trial)
         (dotimes (j n-trial-same-pos)
           (when (or (null pick-result) (eq pick-result :grasp-failed))
-            (let (ec avs)
+            (let (ec avs (offset-z %(-80 + i * -30)))
               (when (eq grasp-style :suction)
-                ;; Fold fingers again
-                (send *ri* :move-hand arm
-                      (send *baxter* :hand-grasp-pose arm :spherical :angle 90) 1000)
                 (setq ec (send *baxter* arm :end-coords :copy-worldcoords))
-                (if (= i 0)
-                  (progn
-                    (send ec :translate #f(0 0 -80) :world)
-                    (pushback (send *baxter* arm :inverse-kinematics ec) avs))
-                  (progn
-                    (send ec :translate #f(0 0 -330) :world)
```

Comment:
-330 is certain value because `:try-to-suction-object-with-gripper-v6` always lift up 300.
Your change will move whole arm when approaching to object.